src/mbgl/text/glyph_atlas.hpp

```diff
@@ -96,16 +78,33 @@ class GlyphAtlas : public util::noncopyable, public GlyphAtlasObserver {
         std::unordered_set<GlyphRequestor*> ids;
     };
 
+    struct GlyphRequest {
+        bool parsed = false;
+        std::unique_ptr<AsyncRequest> req;
+        std::unordered_map<GlyphRequestor*, std::shared_ptr<GlyphDependencies>> requestors;
+    };
+
     struct Entry {
-        std::map<GlyphRange, GlyphPBF> ranges;
+        std::map<GlyphRange, GlyphRequest> ranges;
         GlyphSet glyphSet;
         std::map<uint32_t, GlyphValue> glyphValues;
     };
 
     std::unordered_map<FontStack, Entry, FontStackHash> entries;
-    
-    void removeGlyphValues(GlyphRequestor& requestor, std::map<uint32_t, GlyphValue>& face);
-    void removePendingRanges(GlyphRequestor& requestor, std::map<GlyphRange, GlyphPBF>& ranges);
+
+    // Only used by GlyphAtlasTest
+    friend class ::GlyphAtlasTest;
```

Comment:
While we're moving these, we should probably group `getGlyphSet` here too, it's only used by `GlyphAtlasTest`